php/class-ee-db.php

```diff
@@ -2,325 +2,482 @@
+			EE::error( $exception->getMessage() );
+		}
+
+		$query = 'CREATE TABLE sites (
+			id                   INTEGER  NOT NULL,
+			site_url             VARCHAR  NOT NULL,
+			site_type            VARCHAR  NOT NULL,
+			site_fs_path         VARCHAR  NOT NULL,
+			site_enabled         BOOLEAN  NOT NULL DEFAULT 1,
+			site_ssl             VARCHAR,
+			cache_nginx_browser  BOOLEAN  NOT NULL DEFAULT 0,
+			cache_nginx_fullpage BOOLEAN  NOT NULL DEFAULT 0,
+			cache_mysql_query    BOOLEAN  NOT NULL DEFAULT 0,
+			cache_app_object     BOOLEAN  NOT NULL DEFAULT 0,
+			php_version          VARCHAR,
+			db_name              VARCHAR,
+			db_user              VARCHAR,
+			db_password          VARCHAR,
+			db_root_password     VARCHAR,
+			db_host              VARCHAR,
+			db_port              VARCHAR,
+			app_sub_type         VARCHAR,
+			app_admin_url        VARCHAR,
+			app_admin_email      VARCHAR,
+			app_admin_username   VARCHAR,
+			app_admin_password   VARCHAR,
+			app_mail             VARCHAR,
+			admin_tools          BOOLEAN  NOT NULL DEFAULT 0,
+			created_on           DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			modified_on          DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			site_auth_scope      VARCHAR,
+			site_auth_username   VARCHAR,
+			site_auth_password   VARCHAR,
+			PRIMARY KEY (id),
+			UNIQUE (site_url),
+			CHECK (site_enabled IN (0, 1))
+		);';
+
+		$query .= 'CREATE TABLE site_meta (
+			id INTEGER,
+			site_id INTEGER NOT NULL,
+			meta_key VARCHAR,
+			meta_value VARCHAR,
 			PRIMARY KEY (id),
-			UNIQUE (sitename),
-			CHECK (is_enabled IN (0, 1)),
-			CHECK (is_ssl IN (0, 1))
-		);";
+			FOREIGN KEY (site_id) REFERENCES sites(id)
+		);';
 
-		$query .= "CREATE TABLE migrations (
+		$query .= 'CREATE TABLE migrations (
+			id INTEGER,
 			migration VARCHAR,
-			timestamp DATETIME
-		);";
-
-		$query .= "CREATE TABLE services (
-			id INTEGER NOT NULL,
-			sitename VARCHAR,
-			phpmyadmin BOOLEAN DEFAULT 0,
-			mailhog BOOLEAN DEFAULT 0,
-			postfix BOOLEAN DEFAULT 0,
-			phpredisadmin BOOLEAN DEFAULT 0,
-			adminer BOOLEAN DEFAULT 0,
-			anemometer BOOLEAN DEFAULT 0,
-			debug BOOLEAN DEFAULT 0,
-			PRIMARY KEY (id),
-			FOREIGN KEY (id) REFERENCES sites(id)
-		);";
+			timestamp DATETIME,
+			PRIMARY KEY (id)
+		);';
 
-		$query .= "CREATE TABLE cron (
-			id INTEGER PRIMARY KEY AUTOINCREMENT,
-			sitename VARCHAR,
+		$query .= 'CREATE TABLE cron (
+			id INTEGER,
+			site_url VARCHAR,
 			command VARCHAR,
-			schedule VARCHAR
-		);";
+			schedule VARCHAR,
+			PRIMARY KEY (id),
+			FOREIGN KEY (site_url) REFERENCES sites(site_url)
+		);';
 
-		self::$db->exec( $query );
+		try {
+			self::$pdo->exec( $query );
+		} catch ( PDOException $exception ) {
+			EE::error( 'Encountered Error while creating table: ' . $exception->getMessage() );
+		}
 	}
 
 	/**
-	 * Insert row in table.
+	 * Fetches first record from current query
 	 *
-	 * @param array $data in key value pair.
+	 * @return array Record
+	 * @throws Exception
```

Comment:
Please move this `@throws` before `@return` in this file or if there are other places in this PR for consistency in some place you have added `@throws` before `@return` and in some place after that. Please add it before `@return`.

ref: https://github.com/EasyEngine/easyengine/pull/1143/files#diff-cfaa88ab8e5a4e9f3e67c13e5d049636R66